windows/src/engine/keyman32/appint/appint.cpp

```diff
@@ -203,3 +203,37 @@ AppIntegration::AppIntegration()
 	hwnd = NULL;
   FShiftFlags = 0;
 }
+
+BOOL ContextItemsFromAppContext(WCHAR const* buf, km_kbp_context_item** outPtr)
+{
+
+  km_kbp_context_item* context_items  = new km_kbp_context_item[wcslen(buf) + 1];
+  // would like to check bad allocation but nothrow not known to this compiler
+  WCHAR const *p = buf;
+  uint8_t contextIndex = 0;
+  while (*p) {
+    if (*p == UC_SENTINEL) {
+      // we know the only uc_sentinel code in the context is code_deadkey, which has only 1 parameter: uc_sentinel code_deadkey <deadkey_id>
+      // setup dead key context item
```

Comment:
Let's add a debug assertion for this

Reply:
Is the `assert(*(p + 1) == CODE_DEADKEY);` not what you meant?

Reply:
Oh, yes, that looks good.